backend/src/batch/processor.py

```
@@ -0,0 +1,202 @@
+    if len(sys.argv) < 3:
+        print("Usage: python batch_processor.py <image_dir> <output_json>")
+        sys.exit(1)
+    
+    image_dir = Path(sys.argv[1])
+    output_path = Path(sys.argv[2])
+    
+    asyncio.run(auto_predict_images(
+        image_dir=image_dir,
+        output_path=output_path,
+        max_concurrent=5  # Process 5 images concurrently
+    ))
```

Ruff (0.12.2)

No newline at end of file

Add trailing newline

(W292)